test/models/PenaltyEntrySchema.test.ts

```diff
@@ -10,240 +9,173 @@ describe('Penalty Details Schema Validation', () => {
+                const result = validator.validate(createModelWithCompanyNumber('SC12$$56'));
+                expect(result).to.deep.equal({
+                    errors: [{
+                        field: 'companyNumber',
+                        text: 'You must enter your full eight character company number'
+                    }]
+                })
             });
-
-
         })
-    })
+    });
 
     describe('Penalty Reference', () => {
+        function createModelWithPenaltyReference(penaltyReference: string): PenaltyReferenceDetails {
+            const validCompanyNumber = 'SC123123';
+            return { penaltyReference, companyNumber: validCompanyNumber }
+        }
+
         describe('Happy path', () => {
             it('should accept leading character', () => {
-
-                const penaltyDetails: PenaltyReferenceDetails = {
-                    penaltyReference: 'Z12345678',
-                    companyNumber: 'NI123123'
-                };
-                const result = validator.validate(penaltyDetails)
-                const expectedResult = {errors: []}
-                expect(result).to.deep.equal(expectedResult)
+                const result = validator.validate(createModelWithPenaltyReference('Z12345678'));
+                expect(result).to.deep.equal({errors: []})
             });
-
-        })
+        });
 
         describe('Bad path', () => {
-
             it('should reject empty field', () => {
-                const penaltyDetails: PenaltyReferenceDetails = {
-                    penaltyReference: '',
-                    companyNumber: 'SC123123'
-                };
-                const result = validator.validate(penaltyDetails)
-                const expectedResult = {
-                    errors:[{field: 'penaltyReference',text : 'You must enter a penalty reference number'}]
-                }
-                expect(result).to.deep.equal(expectedResult)
+                const result = validator.validate(createModelWithPenaltyReference(''));
+                expect(result).to.deep.equal({
+                    errors: [{
+                        field: 'penaltyReference',
+                        text: 'You must enter a penalty reference number'
+                    }]
+                })
             });
 
             it('should reject correct penalty reference in lowercase', () => {
-
-                const penaltyDetails: PenaltyReferenceDetails = {
-                    penaltyReference: 'b12345678',
-                    companyNumber: 'NI123123'
-                };
-                const result = validator.validate(penaltyDetails)
-                const expectedResult = {
-                    errors:[{field: 'penaltyReference', text: 'You must enter your reference number exactly as shown on your penalty notice'}]
-                }
-                expect(result).to.deep.equal(expectedResult)
+                const result = validator.validate(createModelWithPenaltyReference('b12345678'));
```

Comment:
this case should be valid according to the  ticket @akaJSaunders

Reply:
Well spotted

Reply:
Changed in most recent version.